cpp/src/io/parquet/writer_impl.cu

```diff
@@ -278,8 +279,8 @@ size_t column_size(column_view const& column, rmm::cuda_stream_view stream)
     return size_of(column.type()) * column.size();
   } else if (column.type().id() == type_id::STRING) {
     auto const scol = strings_column_view(column);
-    return cudf::detail::get_value<size_type>(scol.offsets(), column.size(), stream) -
-           cudf::detail::get_value<size_type>(scol.offsets(), 0, stream);
+    return cudf::strings::detail::get_offset_value(scol.offsets(), column.size(), stream) -
+           cudf::strings::detail::get_offset_value(scol.offsets(), 0, stream);
```

Comment:
Is it possible that the input column could have been sliced?
If so, then this would be more correct.
```suggestion
    return cudf::strings::detail::get_offset_value(scol.offsets(), column.size() + column.offset(), stream) -
           cudf::strings::detail::get_offset_value(scol.offsets(), column.offset(), stream);
```
Note that if the column has not been sliced then `column.offset()==0`.

Reply:
Yes, IIRC it was originally written that way due to a concern @vuule  had about sliced columns.

Reply:
Doesn't `get_offset_value` already adjust for the column offset? AFAICT, it's using `get_value`, which uses `data()`, which is implemented as `head<T>() + _offset`.

Reply:
No. The `get_offset_value()` takes an offsets column which does not include it's parent's sliced values offset/size.

Reply:
Ah, thank you for clearing that up.